Cogs/ClassManagement.py

```diff
@@ -0,0 +1,256 @@
+    @commands.has_permissions(administrator=True)
+    async def destroycourses(self, ctx):
+        """Destroy course channels
+
+        Try loading in cached classlist csv
+        Create list of all channels to be deleted, send to author
+        Get confirmation from author
+        Delete all listed categories and channels
+        Create list of all roles to be deleted, send to author
+        Get confirmation from author
+        Delete all listed roles
+        """
+
+        # Load roles csv
+        csv_filepath = f'role_lists/roles_{ctx.guild.id}.csv'
+
+        # If the role_lists directory doesn't exist raise a FileNotFoundError back to the buildserver command
+        try:
+            courses_df = pd.read_csv(csv_filepath)
+        except FileNotFoundError:
+            raise FileNotFoundError
+
+        # drop cross-listed courses (and other roles if on the file)
+        courses_df = courses_df.dropna(subset=['create_channels'])
+
+        # List of names of categories to be destroyed, as determined by saved csv
+        destroy_category_names = courses_df['text'].tolist()
+        destroy_categories = await self.get_category(ctx, destroy_category_names)
+
+        # List of names of role to be destroyed, as determined by saved csv
+        destroy_role_names = courses_df['role/link'].tolist()
+        destroy_roles = await self.get_roles(ctx, destroy_role_names)
+        
+
+        message = '__**DESTROY FOLLOWING CATEGORIES**__\n'
+        if len(destroy_categories):
+            for category in destroy_categories:
+                message += f'{category.name}\n'
+        else:
+            message += f'NO CATEGORIES FOUND\n'
+        
+        message += '__**DESTROY FOLLOWING ROLES**__\n'
+        if len(destroy_roles):
+            for role in destroy_roles:
+                message += f'{role.mention}\n'
+        else:
+            message += f'NO ROLES FOUND\n'
+        
+        await ctx.send(message)
+        if not await confirmation(self.bot, ctx, 'destroy'):
+            return
+        
+        # Destroy categories and all subchannels
+        for category in destroy_categories:
+            for channel in category.channels:
+                await channel.delete()
+            await category.delete()
+        
+        for role in destroy_roles:
+            await role.delete()
+
+        await ctx.send('***CATEGORIES AND ROLES HAVE BEEN DESTROYED***')
+
+    @commands.command()
+    @commands.has_permissions(administrator=True)
+    async def buildrolemenu(self, ctx):
+        """Creates role menus
+
+        find csv and extracts columns needed through a panda dataframe
+        create the buttons and put them in a view
+        aka the role menu, and send to user
+        """
+
+        # finds csv and extracts appropriate columns using a dataframe
+        csv_filepath = f'role_lists/roles_{ctx.guild.id}.csv'
+        courses_df = pd.read_csv(csv_filepath)
+        short_names = courses_df["text"].to_list()
```

Comment:
Once again, https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.itertuples.html

I'm not sure if it's the best way to do it, but it'll probably look cleaner.

Also, it'd be good to get consistent names for the columns. In fact, once we find some consistent names, we should change the column names in the .csv

Reply:
Issue #151